migrate etrecord generation after to_edge_transform_and_lower to new infra

Gasoonjia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #13151
* #13143
* #13148
* __->__ #13058

Differential Revision: [D79420502](https://our.internmc.facebook.com/intern/diff/D79420502/)

Gasoonjia committed:
migrate etrecord generation after to_edge_transform_and_lower to new infra



Differential Revision: [D79420502](https://our.internmc.facebook.com/intern/diff/D79420502/)

[ghstack-poisoned]

digantdesai commented on examples/xnnpack/aot_compiler.py:
```diff
@@ -103,18 +101,16 @@
             _check_ir_validity=False if args.quantize else True,
             _skip_dim_order=True,  # TODO(T182187531): enable dim order in xnnpack
         ),
+        generate_etrecord=args.etrecord is not None,
```
Add default as `False` above 
```suggestion
        generate_etrecord=args.etrecord is not False,
```

Gasoonjia committed:
Update on "migrate etrecord generation after to_edge_transform_and_lower to new infra"


Differential Revision: [D79420502](https://our.internmc.facebook.com/intern/diff/D79420502/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "migrate etrecord generation after to_edge_transform_and_lower to new infra"


Differential Revision: [D79420502](https://our.internmc.facebook.com/intern/diff/D79420502/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "migrate etrecord generation after to_edge_transform_and_lower to new infra"


Differential Revision: [D79420502](https://our.internmc.facebook.com/intern/diff/D79420502/)

[ghstack-poisoned]
